### DISCUSSION TITLE
H3HexagonLayer data load without updating whole layer, just add hexagons to existing array

### DISCUSSION THREAD

**Discussion by iPOiZN** [visgl/deck.gl#8692] [Q&A]
![deck gl](https://github.com/visgl/deck.gl/assets/94292771/10282e2e-3753-4540-b29d-66b6fac1f7a8)

I'm using H3HexagonLayer in my Vue.js application
I need to do a similar hex loading effect. At the moment I have to refresh the layer every time new data is loaded in this way:

```
this.dataArr = [...this.dataArr, ...response.hexagons.items];
this.$layer.setProps({ data: this.dataArr });
```

How can I make it so that when I add new data to this.dataArr array, I don't have to update the layer?

**Comment by Pessimistress** [comment]
Use [_dataDiff](https://deck.gl/docs/api-reference/core/layer#_datadiff):

```js
new H3HexagonLayer({
    data: this.dataArr.concat(response.hexagons.items),
    _dataDiff: (newData, oldData) => [{startRow: oldData.length, endRow: newData.length}],
    ...
  })
```
